lib/identity_doc_auth/mock/result_response_builder.rb

```diff
@@ -49,7 +49,12 @@ def errors
 
       def parsed_yaml_from_uploaded_file
         @parsed_yaml_from_uploaded_file ||= begin
-          YAML.safe_load(uploaded_file)
+          data = YAML.safe_load(uploaded_file)
+          if data.kind_of?(Hash)
+            data
+          else
+            { 'friendly_error' => "YAML data should have been a hash, got #{data.class}" }
```

Comment:
Should we want this to return an error? My understanding is that we expect `parsed_yaml_from_uploaded_file` to return `nil` in almost all cases except when we're uploading a YAML file, and in all other cases the class's `call` would return a successful response with `DEFAULT_PII_FROM_DOC`. If the file value is a data URL, then I'd expect a `nil` return here as well, unless there's something abnormal about this that we'd want to catch.

Reply:
My thinking was, if we're getting something that parses as valid YAML, but isn't a hash, it's some sort of text file that it shouldn't be, so we want the error?

Another idea I had just now, if we wanted to confirm or disprove the data:// URI theory, is we could also attempt data as an image URI and see what that turns out like? We could consider a valid image something that would return `DEFAULT_PII_FROM_DOC`

Reply:
> Another idea I had just now, if we wanted to confirm or disprove the data:// URI theory, is we could also attempt data as an image URI and see what that turns out like? We could consider a valid image something that would return `DEFAULT_PII_FROM_DOC`

I think that makes sense, and to my original comment, my expectation is that we could be sending data URLs here, and that those should be treated as valid. Though I'm a bit confused how this only appears to be a problem now, since we haven't changed the format of the values we're passing for quite some time. And how it only appears to affect me (as far as I know)?

🤷

Reply:
Don't feel strongly, but shouldn't strings be valid here if they'd be valid to the live SDK?

Reply:
My thinking is, something is probably wrong if we're receiving two different types of data, when the underlying API only accepts one kind of data

But for the time being I'll update this to allow
- Valid YAML hash
- Valid data URI

But a string that is neither of those...still an error, right?

Reply:
that sounds right to me, yeah

Reply:
Yeah, I think that seems reasonable.

Reply:
963df40c47ccec11a5cbd0c233b6c66700cdd560